<launch>
  <!-- Arguments -->
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find multi_allocation)/maps/map.yaml"/>
  <arg name="open_rviz" default="false"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3 -->
  <!--<include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include> -->

  <group ns="tb3_0">
  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL -->
  <include file="$(find multi_allocation)/launch/amcl_0.launch"/>

  <!-- Additional door map -->
  <include file="$(find multi_allocation)/launch/map_updater_node.launch"/>

  <!-- move_base -->
  <include file="$(find multi_allocation)/launch/move_base_aco_0.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>
  <include file="$(find multi_allocation)/launch/dealer_node.launch"/>
  <include file="$(find multi_allocation)/launch/door_scanner_node.launch"/>
  <include file="$(find multi_allocation)/launch/controller_node.launch"/>
  </group>

  <!-- rviz -->
  <group if="$(arg open_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>
</launch>